addons/web_tip/static/src/js/tip.js

```diff
@@ -135,187 +107,231 @@ var Tip = Class.extend({
-                if (!tip.type) {
-                    return true;
-                }
-                return tip.type === type;
-            });
-        }
 
         valid_tips = _.uniq(valid_tips.concat(tips));
         _.each(valid_tips, function(tip) {
-            if (!tip.is_consumed) {
-                self.add_tip(tip);
-            }
+            var t = new Tip(tip);
+            t.do_tip();
         });
     },
 
+    remove_tips: function() {
+        $('.oe_breathing').remove();
+    },
+});
 
-    add_tip: function(tip) {
-        var self = this;
-        self.tip_mutex.exec(function() {
-            if (!tip.is_consumed) {
-                return $.when(self.do_tip(tip));
-            }
-        });
+var Tip = Class.extend({
+    init: function(tip) {
+        this.tip = tip;
+
+        this.$element = null;
+        this.$breathing = null;
+        this.$helper = null;
+        this.$overlay = null;
+        this.$cross = null;
+
+        this.highlight_selector = tip.highlight_selector;
+        this.end_selector = tip.end_selector ? tip.end_selector : tip.highlight_selector;
+        this.triggers = tip.trigger_selector ? tip.trigger_selector.split(',') : [];
     },
 
-    do_tip: function (tip) {
+    do_tip: function() {
         var self = this;
-        var def = $.Deferred();
-        var Tips = new Model('web.tip');
-        var highlight_selector = tip.highlight_selector;
-        var triggers = tip.trigger_selector ? tip.trigger_selector.split(',') : [];
-        var trigger_tip = true;
 
-        if(!$(highlight_selector).length > 0 || !$(highlight_selector).is(":visible")) {
-            return def.reject();
+        if(!$(this.highlight_selector).length > 0 || !$(this.highlight_selector).is(":visible")) {
+            return false;
         }
-        for (var i = 0; i < triggers.length; i++) {
-            if(!$(triggers[i]).length > 0) {
-                trigger_tip = false;
+
+        for (var i = 0; i < this.triggers.length; i++) {
+            if(!$(this.triggers[i]).length > 0) {
+                return false;
             }
         }
 
-        if (trigger_tip) {
-            self.$element = $(highlight_selector).first();
-            if (self.$element.height() === 0 || self.$element.width() === 0) {
-                var $images = self.$element.find('img');
-                if ($images.length > 0) {
-                    $images.first().load(function() {
-                        self.add_tip(tip);
-                    });
-                }
-                return def.reject();
+        this.$element = $(this.highlight_selector).first();
+        if (utils.float_is_zero(this.$element.height(), 1) || utils.float_is_zero(this.$element.width(), 1)) {
+            var $images = this.$element.find('img');
+            if ($images.length > 0) {
+                $images.first().load(function() {
+                    var t = new Tip(self.tip);
+                    t.do_tip();
```

Comment:
You are already inside a tip, why create a new one with the same values and re-check everything there ?